cmd/libs/go2idl/client-gen/generators/generator-for-clientset.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generators
+
+import (
```

Comment:
> To be honest, another motivation for asking for tests was to see how the functions in this file should be used.
Like I should call Filter() first and then GenerateType() and not the other way round.

That is a framework thing, IMO it should be the job for the framework rather than for each individual generator.

> That is like the integration test

Yes, kind of, but writing unit test for each function would be too trivial.